### PR TITLE
tso: refine the codes of TSO and TSO tests

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

Refine the codes of TSO and TSO tests.

### What is changed and how it works?

* Add and merge some common methods.
* Refine the codes.
* Add and adjust some comments.

### Check List

Tests

- Unit test
- Integration test

### CODE REVIEW

File: server/election/leadership_test.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+func Test(t *testing.T) {
+	TestingT(t)
+}
+
+var _ = Suite(&testLeadershipSuite{})
+
+type testLeadershipSuite struct{}
+
+func newTestSingleConfig() *embed.Config {
```
Comment (rleungx) [inline server/election/leadership_test.go:39]: We have already had the function in other places, can we use it?

File: server/tso/allocator_manager.go [hunk 2]
```diff
@@ -358,24 +355,30 @@ func (am *AllocatorManager) allocatorLeaderLoop(ctx context.Context, allocator *
...
+func longSleep(ctx context.Context, waitStep time.Duration) bool {
+	waitTicker := time.NewTicker(waitStep)
+	defer waitTicker.Stop()
+	select {
+	case <-ctx.Done():
+		return false
+	case <-waitTicker.C:
+		return true
+	}
+}
```
Comment (Yisaer) [inline server/tso/allocator_manager.go:380]: I think using `time.After` is more suitable here.

**Review by Yisaer — COMMENTED** [review]
Rest LGTM

**Review by Yisaer — APPROVED** [review]
LGTM

**Review by nolouch — APPROVED** [review]

**Pull request merged**